### PR TITLE
Add image filter to cache

### PR DESCRIPTION
**Pull request by kerty0**
I primary use pixel art wallpapers, so the default filter for all restored images was annoying.
I modified cache so it stores filter and path to image separated by '\n' (I couldn't think of anything better). It can work with old version of cache if it doesn't contain '\n' using default filter.
I also fixed a crash that occurred when using a solid color on my system (EndevourOS + Hyprland): the drawable surface required RGBA image but was only provided with RGB.
I'm still learning Rust and English is not my native language, so sorry if anything is wrong.

### CODE REVIEW

**Comment by LGFae** [comment]
> It can work with old version of cache if it doesn't contain '\n'

We don't need this. Instead, increase the cargo version from masterV2 to masterV3. The cache will won't consider files with older versions.

**Comment by kerty0** [comment]
Now get_previous_image_path returns error to the old version of cache and users will be forced to set the wallpaper again to update cache.

File: client/src/cli.rs [hunk 1]
```diff
@@ -74,6 +75,20 @@ impl std::str::FromStr for Filter {
...
+impl Display for Filter {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let str = match self {
+            Self::Nearest => "Nearest",
+            Self::Bilinear => "Bilinear",
+            Self::CatmullRom => "CatmullRom",
+            Self::Mitchell => "Mitchell",
+            Self::Lanczos3 => "Lanczos3",
+        }
+        .to_string();
```
Comment (LGFae) [inline client/src/cli.rs:87]: You shouldn't need the to_string call here.

File: common/src/cache.rs [hunk 2]
```diff
@@ -64,28 +64,35 @@ pub fn load_animation_frames(
...
 
-pub fn get_previous_image_path(output_name: &str) -> io::Result<String> {
+pub fn get_previous_image_path(output_name: &str) -> io::Result<(String, String)> {
     let mut filepath = cache_dir()?;
     clean_previous_verions(&filepath);
 
     filepath.push(output_name);
     if !filepath.is_file() {
-        return Ok("".to_string());
+        return Ok(("Lanczos3".to_string(), "".to_string()));
```
Comment (LGFae) [inline common/src/cache.rs:73]: I don't really get what this gives us. Is it just to improve the error messages?

**Review by LGFae — COMMENTED** [review]
Very nice. Just two small things and we can merge this.

**Comment by kerty0** [comment]
How swww restore don't stop half done if one of the outputs don't have cache file and print name of the output for all errors

File: common/src/cache.rs (hunk 2 above)
Comment (kerty0) [reply common/src/cache.rs]: Yes, and it makes it easy to force swww-daemon ignore missing cache files.

**Comment by LGFae** [comment]
Thanks!

**Pull request merged**